ui/app/AppLayouts/Wallet/controls/TokenSelector.qml

```diff
@@ -0,0 +1,212 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+import QtGraphicalEffects 1.0
+
+import StatusQ 0.1
+import StatusQ.Components 0.1
+import StatusQ.Components.private 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Utils 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Popups.Dialog 0.1
+
+import AppLayouts.Wallet.views 1.0
+
+import utils 1.0
+import shared.controls 1.0
+
+ComboBox {
+    id: root
+
+    // expected model structure:
+    // tokensKey, name, symbol, decimals, currencyBalanceAsString (computed), marketDetails, balances -> [ chainId, address, balance, iconUrl ]
+
+    // input API
+    property string nonInteractiveDelegateKey
```

Comment:
on other thoughts maybe this could be part of model and not needed as a separate property? just something to think about.